syftlib/src/test/java/org/openmined/syft/unit/networking/clients/WebSocketClientTest.kt

```diff
@@ -1,4 +1,36 @@
 package org.openmined.syft.unit.networking.clients
 
+import com.nhaarman.mockitokotlin2.mock
+import kotlinx.serialization.json.json
+import org.junit.Test
+import org.openmined.syft.networking.clients.SyftWebSocket
+import org.openmined.syft.networking.requests.NetworkingProtocol
+
+@ExperimentalUnsignedTypes
 internal class WebSocketClientTest {
+
+    private val webSocket = SyftWebSocket(NetworkingProtocol.HTTP, "localhost", 2000u)
+
+    @Test
+    fun `verify calling start function should connect and initialized the web socket`() {
+        val processor = webSocket.start()
+        processor.test().assertEmpty()
+        processor.test().assertNotComplete()
+    }
+
+    @Test
+    fun `verify calling send function should sending message through the socket connection`() {
+        webSocket.start()
+        val message = json {
+            "data" to "data"
+        }
+        assert(webSocket.send(message))
+    }
+
+    @Test
+    fun `verify calling close function should close the socket connection`() {
+        webSocket.start()
+        assert(webSocket.close())
+    }
```

Comment:
Also add the tests to check that the websocket reconnects for n number of times with the appropriate delay